### PR TITLE
fix(PVS/V1048): fix "variable was assigned the same value" warning

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/mpack/conv.c [hunk 1]
```diff
@@ -328,7 +327,6 @@ static mpack_value_t mpack_pack_ieee754(double v, unsigned mantbits,
   double mant;
 
   if (v == 0) {
-    rv.lo = 0;
     rv.hi = 0;
```
Comment (justinmk) [inline src/mpack/conv.c:330]: won't it also complain about rv.hi here?

Typically we leave the declaration uninitialized. That forces the fields to be set explicitly in each codepath.

Comment (dundargoc) [reply src/mpack/conv.c:330]: That's... a good question. I have no idea why PVS didn't warn about `rv.hi`.

Didn't quite follow how I should solve this. Could you elaborate a bit more?

Comment (justinmk) [reply src/mpack/conv.c:330]: My suggestion was to change `mpack_value_t rv = {0, 0};` to just `mpack_value_t rv;` .

Comment (dundargoc) [reply src/mpack/conv.c:330]: Gotcha.

**Comment by dundargoc** [comment]
There was some error with one of the changes so I just opted to skip it :).

**Comment by justinmk** [comment]
> There was some error with one of the changes so I just opted to skip it :).

well, it was indicating that some codepaths didn't initialize all the fields...

**Comment by dundargoc** [comment]
I understand, I just wasn't sure what the preferred way to do that was so I opted to skip it. Should I fix it instead?

**Pull request merged**